File: CIVITAS/src/civitas/PorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class PorCasaHotel  extends Sorpresa{
    PorCasaHotel(Tablero tablero , int valor , String texto){
        super(tablero ,valor, texto , null);
    }
    
    @Override
    void aplicarAJugador(int actual  ,ArrayList<Jugador> todos) {
       if(this.jugadorCorrecto(actual, todos)){
        this.informe(actual, todos);
        Jugador j =  todos.get(actual);
        j.modificarSaldo(this.getValor() * j.cantidadCasasHoteles());
    }
    }
}

File: CIVITASRUBY/lib/por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  
  require_relative 'casilla'
  class Por_casa_hotel < Sorpresa
    def initialize(tablero,valor,texto)
      super(tablero,valor,texto,nil)
    end
    
    def aplicar_a_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].modificar_saldo(@valor.to_f * todos[actual].cantidad_casas_hoteles)
      end
    end
    
  end
end

